### PR TITLE
Sudha ansible

### PR DESCRIPTION
**Pull request by sudhakulkarni123**
new files with all changes

### CODE REVIEW

File: Blue_Green_Ami.pkr.hcl [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+}
+
+source "amazon-ebs" "ubuntu" {
+  ami_name      = "${var.ami_prefix1}-${local.timestamp}"
+  instance_type = "t2.micro"
+  region        = "ap-south-1"
+  vpc_id        = "vpc-0e4a12ff878f00312"
+  subnet_id     = "subnet-0973eb0379fbbe882"
+  security_group_id = "sg-0993a9913a37d35be"
+  // ssh_interface = "private_ip"
```
Comment (pasc-ed) [inline Blue_Green_Ami.pkr.hcl]: Commented lines should be removed if it's not required anymore.

File: Blue_Green_Ami.pkr.hcl [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+      root-device-type    = "ebs"
+      virtualization-type = "hvm"
+    }
+    most_recent = true
+    owners      = ["099720109477"]
+  }
+  ssh_username = "ubuntu"
+}
+
+variable "ami_prefix2" {
```
Comment (pasc-ed) [inline Blue_Green_Ami.pkr.hcl]: For visibility, can you regrouped all variables together in the file please.

File: Blue_Green_Ami.pkr.hcl [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+variable "ami_prefix1" {
+  type    = string
+  default = "Blue_Ami"
+}
+
+locals {
+  timestamp = regex_replace(timestamp(), "[- TZ:]", "")
+}
+
+source "amazon-ebs" "ubuntu" {
```
Comment (pasc-ed) [inline Blue_Green_Ami.pkr.hcl]: Could you provide a proper name for the resource here please, instead of just the `ubuntu` name. E.g. this is for the blue ami, so you could try a name like `blue-web-server`.

File: Blue_Green_Ami.pkr.hcl [hunk 4]
```diff
@@ -0,0 +1,89 @@
+packer {
+  required_plugins {
+    amazon = {
+      version = ">= 1.1.1"
+      source  = "github.com/hashicorp/amazon"
+    }
+  }
+}
+
+variable "ami_prefix1" {
```
Comment (pasc-ed) [inline Blue_Green_Ami.pkr.hcl]: Could you provide a proper name for the variable please, this will have allow for a better debug later.

File: Blue_Green_Ami.pkr.hcl [hunk 5]
```diff
@@ -0,0 +1,89 @@
...
+      virtualization-type = "hvm"
+    }
+    most_recent = true
+    owners      = ["099720109477"]
+  }
+  ssh_username = "ubuntu"
+}
+
+build {
+  name = "learn-packer"
```
Comment (pasc-ed) [inline Blue_Green_Ami.pkr.hcl]: Could you give a proper name for the build as well please.

File: Blue_Green_Ami.pkr.hcl [hunk 6]
```diff
@@ -0,0 +1,89 @@
...
+  sources = [
+    "source.amazon-ebs.ubuntu"
+  ]
+
+  provisioner "ansible" {
+    playbook_file = "./main.yml"
+    extra_arguments = ["--extra-vars", "@/c_blue.yml"]
```
Comment (pasc-ed) [inline Blue_Green_Ami.pkr.hcl]: is this configuration working ? shouldn't you be using the parth to the variable file as well ?

File: templates/index.html.j2 [hunk 7]
```diff
@@ -1,8 +1,8 @@
 <html>
-    <title>{Blue/Green} Application</title>
+    <title>Blue/green Application</title>
```
Comment (pasc-ed) [inline templates/index.html.j2:2]: You should use a variable to define the name of the application. It should be either `Blue` or `Green` not both.

File: user_data.sh [hunk 8]
```diff
@@ -0,0 +1,3 @@
+#!/bin/bash
```
Comment (pasc-ed) [inline user_data.sh]: Why is the user-data file here ? This should be used by the Launch configuration of the Autoscaling group, which is another folder.

File: vars.pckrvars.hcl [hunk 9]
```diff
@@ -0,0 +1,2 @@
+ami_prefix1 = "Blue_Ami"
```
Comment (pasc-ed) [inline vars.pckrvars.hcl]: Is this file required as well since your variable already has a default value to use, in the `Blue_Green_Ami.pkr.hcl` file ?

**Review by pasc-ed — CHANGES_REQUESTED** [review]

File: Blue_Green_Ami.pkr.hcl (hunk 6 above)
Comment (jinendradhurvey) [reply Blue_Green_Ami.pkr.hcl]: ya i have given the path.
